### PR TITLE
check for running apiserver with curl

### PR DESCRIPTION
**Pull request by thz**
**What this PR does / why we need it**:

Adds an enhanced check for running apiserver with curl in a loop.
This allows a trailing dot in service names.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Documentation**:

**Release note**:

```release-note
Improve detection of user-cluster apiserver health on startup.
```

### CODE REVIEW

File: api/pkg/resources/apiserver/is-running.go [hunk 1]
```diff
@@ -20,14 +20,19 @@ func IsRunningInitContainer(data resources.DeploymentDataProvider) (*corev1.Cont
...
-			"--show-error",
-			fmt.Sprintf("%s/healthz", url),
+			"-c",
+			fmt.Sprintf(`
+			name=%s
+			port=%s
+			# manual re-resolving enables use of trailing dot with curl
+			ip=$(getent hosts $name | cut -d" " -f1)
+			while ! curl --resolve $name:$port:$ip --insecure --silent --show-error --max-time 3 %s/healthz; do
+				[ $(( timeout++ )) -gt 100 ] && exit 1
```
Comment (alvaroaleman) [inline api/pkg/resources/apiserver/is-running.go]: Maybe echo an error?

Comment (thz) [reply api/pkg/resources/apiserver/is-running.go]: added verbosity.

File: api/pkg/resources/apiserver/is-running.go [hunk 2]
```diff
@@ -20,14 +20,23 @@ func IsRunningInitContainer(data resources.DeploymentDataProvider) (*corev1.Cont
...
+			port=%s
+			# manual re-resolving enables use of trailing dot with curl
+			ip=$(getent hosts $name | cut -d" " -f1)
+			while ! curl --resolve $name:$port:$ip --insecure --silent --show-error --max-time 3 %s/healthz; do
+				if [ $(( timeout++ )) -gt 100 ]; then
+					echo "Giving up after 100 tries."
+					exit 1
+				fi
+				sleep 2
+				echo "Retry $timeout/3"
```
Comment (alvaroaleman) [inline api/pkg/resources/apiserver/is-running.go]: Why `[...]/3`? Its 100 tries not three?

**Comment by alvaroaleman** [comment]
Release Note is missing btw

File: api/pkg/resources/apiserver/is-running.go (hunk 2 above)
Comment (thz) [reply api/pkg/resources/apiserver/is-running.go]: Some logarithmic scale ;-)
fixing.

**Comment by thz** [comment]
All issues addressed. 
@mrIncompetent needs to codeowner-approve the beast.

**Review by mrIncompetent — APPROVED** [review]

**Pull request merged**